lib/galaxy/managers/workflows.py

```diff
@@ -583,6 +585,106 @@ def _workflow_to_dict_run(self, trans, stored, workflow):
             'workflow_resource_parameters': self._workflow_resource_parameters(trans, stored, workflow),
         }
 
+    def _workflow_to_dict_preview(self, trans, workflow):
+        """
+        Builds workflow dictionary containing input labels and values.
```

Comment:
If this is to be embedded into a workflow report would it make more sense to process a workflow invocation ?

Reply:
I think in many cases it might make sense to display an `Invocation` instead of a `Workflow` but they don't exclude each other. The content of the dictionary generated above is similar to the content which was previously generated by the Workflow `display.mako`.